src/app/hm-admin/services/schema/schema.service.ts

```diff
@@ -0,0 +1,171 @@
+import {Injectable} from '@angular/core';
+import {Http} from '@angular/http';
+import {Observable, BehaviorSubject} from 'rxjs';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/switchMap';
+import 'rxjs/add/operator/share';
+import 'rxjs/add/operator/last';
+import 'rxjs/add/operator/zip';
+import * as _ from 'lodash';
+import {ConfigService} from '../config/Config.service';
+import {entrypointHelper} from '../../helper/entrypointHelper';
+import {schemaHelper} from '../../helper/schemaHelper';
+import {EntryPoint} from '../models/EntryPoint';
+import {stringHelper} from '../../helper/stringHelper';
+import { CoreDefinitionService } from '../core/CoreDefinition.service';
+
+@Injectable()
+export class SchemaService {
+
+  schema: BehaviorSubject<any>;
+  entrypoints: BehaviorSubject<Array<EntryPoint>>;
+  availableEntrypoints: Array<EntryPoint> = [];
+
+  constructor(private http: Http,
+              private config: ConfigService,
+              private coreDefinitionService: CoreDefinitionService) {
+    this.schema = new BehaviorSubject(null);
+    this.entrypoints = new BehaviorSubject([]);
+
+  }
+
+  /**
+   * load API description
+   */
+  load() {
+    this.coreDefinitionService.getDefinitions().subscribe((definition) => {
+      this.loadRoot();
+      this.loadDocumentation(definition);
+    });
+
+  }
+
+  /**
+   * load available entrypoints from API's root
+   * @private
+   */
+  loadRoot() {
+    let request: string = this.config.api.baseUrl;
+    let entrypoints = this.http.get(request)
+      .map(data => data.json())
+      .map(entrypointHelper.filterEntryPoints);
+
+    entrypoints.subscribe(
+      (entrypoints => {
+        _.forEach(_.keys(entrypoints), (key) => {
+          let cleanModel = stringHelper.toCamelCase(entrypointHelper.getModelById(key)),
+            entrypoint: EntryPoint = entrypointHelper.getEntryPointByModel(key, this.availableEntrypoints);
+
+          if (undefined === entrypoint) {
+            entrypoint = new EntryPoint(cleanModel);
+            this.availableEntrypoints.push(entrypoint);
+          }
+
+          entrypoint.url = entrypoints[key];
+        });
+        this.entrypoints.next(this.availableEntrypoints);
+      }),
+      () => {
+        throw new Error(`API's url is unreachable: ${request}.`);
```

Comment:
We are not, if the API is really unreachable, you will get this error. Trust me with that :p